src/Microsoft.Framework.Logging.Console/ConsoleLoggerFactoryExtensions.cs

```diff
@@ -35,5 +37,23 @@ public static ILoggerFactory AddConsole(this ILoggerFactory factory, LogLevel mi
             factory.AddProvider(new ConsoleLoggerProvider((category, logLevel) => logLevel >= minLevel));
             return factory;
         }
+
+        /// <summary>
+        ///     Adds a console logger that is enabled as defined by the IConfiguration object
+        /// </summary>
+        /// <param name="configuration">IConfiguration object with filter attributes</param>
+        /// <returns></returns>
+        public static ILoggerFactory AddConsole(this ILoggerFactory factory, IConfiguration configuration)
+        {
+            factory.AddProvider(new ConsoleLoggerProvider((category, logLevel) =>
+            {
+                LogLevel minLevel;
+
+                return Enum.TryParse(configuration?.GetSubKey("ConsoleLogger")?.Get(category), out minLevel) &&
```

Comment:
You'd presumably want to do this just once, when the AddConsole method is called. That way you don't parse the enum every time, but more importantly, you fail fast in case there's some funky config. Otherwise the failure won't happen until super late (and ironically, probably won't get logged because the logger is busted!).